build/submodules.props

```diff
@@ -2,59 +2,89 @@
+
+          CascadeVersion
+            Produce new package versions if there were changes to product code, or if one of the package dependencies has updated.
+            Examples: metapackages which are not top-level, but should still be used to help users get the latest transitive set of dependencies
+
+          AlwaysUpdate
+            Packages should update in every patch.
+            Examples: top-level metapackages and templates.
+
+       -->
+      <PatchPolicy>ProductChangesOnly</PatchPolicy>
     </Repository>
+    <ShippedRepository>
+      <Build>false</Build>
+      <PatchPolicy>ProductChangesOnly</PatchPolicy>
+    </ShippedRepository>
   </ItemDefinitionGroup>
 
+  <PropertyGroup>
+    <TestsRequiredTheSharedRuntime Condition="'$(RepositoryToBuild)' == 'Templating'">true</TestsRequiredTheSharedRuntime>
+  </PropertyGroup>
+
   <ItemGroup>
-    <Repository Include="AADIntegration" />
-    <Repository Include="Antiforgery" />
-    <Repository Include="AuthSamples" />
-    <Repository Include="AzureIntegration" />
-    <Repository Include="BasicMiddleware" />
-    <Repository Include="BrowserLink" />
     <Repository Include="Caching" />
     <Repository Include="Common" />
-    <Repository Include="Configuration" />
-    <Repository Include="CORS" />
-    <Repository Include="DataProtection" />
-    <Repository Include="DependencyInjection" />
-    <Repository Include="Diagnostics" />
-    <Repository Include="DotNetTools" />
     <Repository Include="EntityFrameworkCore" />
-    <Repository Include="FileSystem" />
-    <Repository Include="Hosting" />
-    <Repository Include="HtmlAbstractions" />
-    <Repository Include="HttpAbstractions" />
-    <Repository Include="HttpClientFactory" />
-    <Repository Include="HttpSysServer" />
     <Repository Include="Identity" />
-    <Repository Include="IISIntegration" />
-    <Repository Include="JavaScriptServices" />
-    <Repository Include="JsonPatch" />
     <Repository Include="KestrelHttpServer" />
-    <Repository Include="Localization" />
-    <Repository Include="Logging" />
-    <Repository Include="MetaPackages" />
-    <Repository Include="MusicStore" />
     <Repository Include="Mvc" />
-    <Repository Include="MvcPrecompilation" />
-    <Repository Include="Options" />
     <Repository Include="Razor" />
-    <Repository Include="ResponseCaching" />
-    <Repository Include="Routing" />
-    <Repository Include="Scaffolding" />
-    <Repository Include="Security" />
-    <Repository Include="ServerTests" />
-    <Repository Include="Session" />
     <Repository Include="SignalR" />
-    <Repository Include="StaticFiles" />
-    <Repository Include="WebSockets" />
-    <!-- <Repository Include="Templating" /> -->
+    <Repository Include="Scaffolding" PatchPolicy="AlwaysUpdate" />
+    <Repository Include="MetaPackages" PatchPolicy="CascadeVersions" />
+    <Repository Include="Templating" PatchPolicy="AlwaysUpdateAndCascadeVersions" />
```

Comment:
Care to explain why `AlwaysUpdateAndCascadeVersions` exists? Are `CascadeVersions` and `AlwaysUpdate` two independent concepts?

Reply:
Scaffolding = AlwaysUpdate. Due to the way they've design VS, this needs to ship whenever aspnetcore.app does, *BUT* it doesn't need to update its dependencies to latest

MetaPackages = CascadeVersions. If there are changes to its dependencies, it should reship because it should help nesure transitive versions are latest

Templating = both because should be true. Use latest, and always update.

They could be independent vars, but I got tired of having dozens of boolean flags